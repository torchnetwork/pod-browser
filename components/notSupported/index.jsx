/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { Container, Content, PageHeader } from "@inrupt/prism-react-components";
import React from "react";
import { useRouter } from "next/router";

export const TESTCAFE_ID_NOT_SUPPORTED = "not-supported";

export default function NotSupported() {
  const router = useRouter();
  return (
    <div data-testid={TESTCAFE_ID_NOT_SUPPORTED}>
      <PageHeader title="Resource Not Supported" />
      <Container>
        <Content>
          <p>PodBrowser can only show resources that are Solid containers.</p>
          <p>
            <span>
              Navigate to the resource directly (this will navigate away from
              PodBrowser):&nbsp;
            </span>
            <a href={router.query.iri}>{router.query.iri}</a>
          </p>
        </Content>
      </Container>
    </div>
  );
}
