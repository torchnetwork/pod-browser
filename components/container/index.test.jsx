/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React from "react";
import * as RouterFns from "next/router";
import { mockSolidDatasetFrom } from "@inrupt/solid-client";
import { renderWithTheme } from "../../__testUtils/withTheme";
import Container from "./index";
import useContainerResourceIris from "../../src/hooks/useContainerResourceIris";
import useDataset from "../../src/hooks/useDataset";
import useAuthenticatedProfile from "../../src/hooks/useAuthenticatedProfile";
import { mockProfileAlice } from "../../__testUtils/mockPersonResource";
import usePodRootUri from "../../src/hooks/usePodRootUri";
import useResourceInfo from "../../src/hooks/useResourceInfo";
import useAccessControl from "../../src/hooks/useAccessControl";
import { TESTCAFE_ID_AUTH_PROFILE_LOAD_ERROR } from "../authProfileLoadError";
import { TESTCAFE_ID_POD_ROOT_LOAD_ERROR } from "../podRootLoadError";
import { TESTCAFE_ID_NO_CONTROL_ERROR } from "../noControlWarning";
import { TESTCAFE_ID_ACCESS_FORBIDDEN } from "../accessForbidden";
import { TESTCAFE_ID_RESOURCE_NOT_FOUND } from "../resourceNotFound";
import { TESTCAFE_ID_NOT_SUPPORTED } from "../notSupported";

jest.mock("../../src/hooks/useDataset");
jest.mock("../../src/hooks/useContainerResourceIris");
jest.mock("../../src/hooks/useAuthenticatedProfile");
jest.mock("../../src/hooks/usePodRootUri");
jest.mock("../../src/hooks/useResourceInfo");
jest.mock("../../src/hooks/useAccessControl");

describe("Container view", () => {
  const iri = "https://example.com/container/";
  const container = mockSolidDatasetFrom(iri);

  beforeEach(() => {
    useDataset.mockReturnValue({ data: container });
    useAuthenticatedProfile.mockReturnValue({
      data: mockProfileAlice(),
    });
    usePodRootUri.mockReturnValue(iri);
    useResourceInfo.mockReturnValue({ data: container });
    useAccessControl.mockReturnValue({});
    jest.spyOn(RouterFns, "useRouter").mockReturnValue({
      asPath: "asPath",
      replace: jest.fn(),
      query: {},
    });
    useContainerResourceIris.mockReturnValue({
      data: [
        "https://myaccount.mypodserver.com/inbox",
        "https://myaccount.mypodserver.com/private",
        "https://myaccount.mypodserver.com/note.txt",
      ],
      mutate: () => {},
    });
  });

  it("renders a table", () => {
    const { asFragment } = renderWithTheme(<Container iri={iri} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Renders a spinner if data is loading", () => {
    useContainerResourceIris.mockReturnValue({
      data: undefined,
      mutate: () => {},
    });

    const { asFragment } = renderWithTheme(<Container iri={iri} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Access Forbidden if the fetch for container returns 401", () => {
    useDataset.mockReturnValue({
      error: { message: "401" },
    });
    const { asFragment, getByTestId } = renderWithTheme(
      <Container iri={iri} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTCAFE_ID_ACCESS_FORBIDDEN)).toBeDefined();
  });

  it("renders Access Forbidden if the fetch for container returns 403", () => {
    useDataset.mockReturnValue({
      error: { message: "403" },
    });
    const { asFragment, getByTestId } = renderWithTheme(
      <Container iri={iri} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTCAFE_ID_ACCESS_FORBIDDEN)).toBeDefined();
  });

  it("renders Access Forbidden if the fetch for container returns 404", () => {
    useDataset.mockReturnValue({
      error: { message: "404" },
    });
    const { asFragment, getByTestId } = renderWithTheme(
      <Container iri={iri} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTCAFE_ID_RESOURCE_NOT_FOUND)).toBeDefined();
  });

  it("renders Not supported if the fetch for container returns 500", () => {
    useDataset.mockReturnValue({
      error: { message: "500" },
    });
    const { asFragment, getByTestId } = renderWithTheme(
      <Container iri={iri} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTCAFE_ID_NOT_SUPPORTED)).toBeDefined();
  });

  it("renders Not Supported if resource loaded is not a container", () => {
    const resourceIri = "http://example.com/resource";
    useDataset.mockReturnValue({
      data: mockSolidDatasetFrom(resourceIri),
    });
    const { asFragment } = renderWithTheme(<Container iri={resourceIri} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders AuthProfileLoadError if fetching authenticated profile fails", () => {
    useAuthenticatedProfile.mockReturnValue({ error: new Error() });
    const { asFragment, getByTestId } = renderWithTheme(
      <Container iri={iri} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTCAFE_ID_AUTH_PROFILE_LOAD_ERROR)).toBeDefined();
  });

  it("renders PodRootLoadError if it fails to fetch resourceInfo for podRoot", () => {
    useResourceInfo.mockReturnValue({ error: new Error() });
    const { asFragment, getByTestId } = renderWithTheme(
      <Container iri={iri} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTCAFE_ID_POD_ROOT_LOAD_ERROR)).toBeDefined();
  });

  it("renders NoControlError if it fails to get access control for podRoot", () => {
    useAccessControl.mockReturnValue({ error: new Error() });
    const { asFragment, getByTestId } = renderWithTheme(
      <Container iri={iri} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTCAFE_ID_NO_CONTROL_ERROR)).toBeDefined();
  });
});
