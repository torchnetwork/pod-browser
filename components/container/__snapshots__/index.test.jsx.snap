// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container view Renders a spinner if data is loading 1`] = `
<DocumentFragment>
  <div
    class="PodBrowser-loading-indicator-container PodBrowser-loading-indicator-container--center"
  >
    <svg
      class="PodBrowser-loading-indicator"
      viewBox="22 22 44 44"
    >
      <circle
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`Container view renders Access Forbidden if the fetch for container returns 401 1`] = `
<DocumentFragment>
  <div
    data-testid="access-forbidden"
  >
    <div
      class="PodBrowser-page-header"
    >
      <h1
        class="PodBrowser-page-header__title"
      >
        No Access
      </h1>
      <div
        class="PodBrowser-page-header__content"
      />
      <div
        class="PodBrowser-page-header__toolbar"
      />
    </div>
    <div
      class="PodBrowser-container"
    >
      <p>
        You don't have access to this resource: 
        <a
          href="/resource/[iri]"
        />
        .
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container view renders Access Forbidden if the fetch for container returns 403 1`] = `
<DocumentFragment>
  <div
    data-testid="access-forbidden"
  >
    <div
      class="PodBrowser-page-header"
    >
      <h1
        class="PodBrowser-page-header__title"
      >
        No Access
      </h1>
      <div
        class="PodBrowser-page-header__content"
      />
      <div
        class="PodBrowser-page-header__toolbar"
      />
    </div>
    <div
      class="PodBrowser-container"
    >
      <p>
        You don't have access to this resource: 
        <a
          href="/resource/[iri]"
        />
        .
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container view renders Access Forbidden if the fetch for container returns 404 1`] = `
<DocumentFragment>
  <div
    data-testid="resource-not-found"
  >
    <div
      class="PodBrowser-page-header"
    >
      <h1
        class="PodBrowser-page-header__title"
      >
        Resource Not Found
      </h1>
      <div
        class="PodBrowser-page-header__content"
      />
      <div
        class="PodBrowser-page-header__toolbar"
      />
    </div>
    <div
      class="PodBrowser-container"
    >
      <p>
        Unable to find the resource: 
        <a
          href="/resource/[iri]"
        />
        .
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container view renders AuthProfileLoadError if fetching authenticated profile fails 1`] = `
<DocumentFragment>
  <div
    data-testid="auth-profile-load-error"
  >
    <div
      class="PodBrowser-page-header"
    >
      <h1
        class="PodBrowser-page-header__title"
      >
        Failed loading your profile
      </h1>
      <div
        class="PodBrowser-page-header__content"
      />
      <div
        class="PodBrowser-page-header__toolbar"
      />
    </div>
    <div
      class="PodBrowser-container"
    >
      <p>
        PodBrowser were not able to load your profile.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container view renders NoControlError if it fails to get access control for podRoot 1`] = `
<DocumentFragment>
  <div
    class="PodBrowser-page-header PodBrowser-files-container-header"
  >
    <h1
      class="PodBrowser-page-header__title"
    >
      Files
    </h1>
    <div
      class="PodBrowser-page-header__content"
    />
    <div
      class="PodBrowser-page-header__toolbar"
    />
  </div>
  <div
    class="PodBrowser-drawer-container"
  >
    <div
      class="PodBrowser-drawer-container__content"
    >
      <div
        class="PodBrowser-sub-header"
      >
        <div
          class="PodBrowser-loading-indicator-container PodBrowser-loading-indicator-container--center"
        >
          <svg
            class="PodBrowser-loading-indicator"
            viewBox="22 22 44 44"
          >
            <circle
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
        <div
          class="PodBrowser-actions-container"
        >
          <a
            class="PodBrowser-page-header__action"
            data-testid="folder-details-button"
            href="/resource/[iri]?action=details&resourceIri="
          >
            Folder Details
          </a>
          <button
            class="PodBrowser-page-header__action"
            data-testid="add-folder-button"
            type="button"
          >
            Create Folder
          </button>
          <label
            class="PodBrowser-page-header__action"
            data-testid="upload-file-button"
            for="upload-file-input"
            tabindex="0"
          >
            Upload File
            <input
              data-testid="upload-file-input"
              id="upload-file-input"
              style="display: none;"
              type="file"
            />
          </label>
        </div>
      </div>
      <div
        class="PodBrowser-message PodBrowser-message--warning PodBrowser-message--prominent"
        data-testid="no-control-error"
      >
        <div
          class="PodBrowser-message__icon PodBrowser-message__icon--warning"
        >
          <span
            class="PodBrowser-icon-error"
          />
        </div>
        <div
          class="PodBrowser-message__main"
        >
          <p
            class="PodBrowser-message__title"
          >
            You can't share anything or change permissions in your Pod
          </p>
          <p>
            This is because you don't have control access. You can still browse the resources in your Pod, but you won't be able to share anything.
          </p>
          <p
            class="PodBrowser-message__title"
          >
            How to fix:
          </p>
          <ol>
            <li>
              <span>
                Log in to 
              </span>
              <a
                href="https://example.com/container/"
                rel="noreferrer"
                target="_blank"
              >
                <span>
                  your Pod
                </span>
                <span
                  class="PodBrowser-icon-external-link"
                  data-external="true"
                />
              </a>
            </li>
            <li>
              Go to “Preferences”
            </li>
            <li>
              <span>
                Under “Manage your trusted applications,” look for 
              </span>
              <span>
                http://localhost
              </span>
              <span>
                 and check the “Control” box
              </span>
            </li>
            <li>
              Click on “Update”
            </li>
            <li>
              Return to PodBrowser
            </li>
          </ol>
        </div>
        <div
          class="PodBrowser-message__close PodBrowser-message__close--warning"
        >
          <button
            class="PodBrowser-button PodBrowser-button--text"
            data-testid="no-control-close-button"
          >
            <span
              aria-label="Close message"
              class="PodBrowser-icon-cancel"
              role="img"
            />
          </button>
        </div>
      </div>
      <table
        class="PodBrowser-table"
        role="table"
      >
        <thead
          class="PodBrowser-table__header"
        >
          <tr
            class="PodBrowser-table__header-row"
            role="row"
          >
            <td
              class="PodBrowser-table__header-cell PodBrowser-table__header-cell--align-center PodBrowser-table__header-cell--width-preview"
              colspan="1"
              role="columnheader"
            >
                
            </td>
            <td
              class="PodBrowser-table__header-cell PodBrowser-table__header-cell--align-center PodBrowser-table__header-cell--width-preview"
              colspan="1"
              role="columnheader"
            >
                
            </td>
            <td
              class="PodBrowser-table__header-cell"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              File 
            </td>
            <td
              class="PodBrowser-table__header-cell"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              Type 
            </td>
          </tr>
        </thead>
        <tbody
          class="PodBrowser-table__body"
          role="rowgroup"
        >
          <tr
            class="PodBrowser-table__body-row PodBrowser-table__body-row--selectable"
            tabindex="0"
          >
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <button
                class="PodBrowser-button PodBrowser-button--text PodBrowser-bookmark-button"
                data-testid="bookmark-button"
                disabled=""
                type="button"
              >
                <i
                  alt="bookmark button"
                  aria-label="bookmark"
                  class="PodBrowser-icon-star PodBrowser-bookmark-icon-unselected"
                />
              </button>
            </td>
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <i
                class="PodBrowser-icon-file PodBrowser-resource-icon"
              />
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              inbox
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              Resource
            </td>
          </tr>
          <tr
            class="PodBrowser-table__body-row PodBrowser-table__body-row--selectable"
            tabindex="0"
          >
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <button
                class="PodBrowser-button PodBrowser-button--text PodBrowser-bookmark-button"
                data-testid="bookmark-button"
                disabled=""
                type="button"
              >
                <i
                  alt="bookmark button"
                  aria-label="bookmark"
                  class="PodBrowser-icon-star PodBrowser-bookmark-icon-unselected"
                />
              </button>
            </td>
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <i
                class="PodBrowser-icon-file PodBrowser-resource-icon"
              />
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              private
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              Resource
            </td>
          </tr>
          <tr
            class="PodBrowser-table__body-row PodBrowser-table__body-row--selectable"
            tabindex="0"
          >
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <button
                class="PodBrowser-button PodBrowser-button--text PodBrowser-bookmark-button"
                data-testid="bookmark-button"
                disabled=""
                type="button"
              >
                <i
                  alt="bookmark button"
                  aria-label="bookmark"
                  class="PodBrowser-icon-star PodBrowser-bookmark-icon-unselected"
                />
              </button>
            </td>
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <i
                class="PodBrowser-icon-file PodBrowser-resource-icon"
              />
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              note.txt
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              Resource
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <section
      class="PodBrowser-drawer"
    >
      <div
        class="PodBrowser-drawer__content"
      >
        <div
          class="PodBrowser-drawer__content-wrapper"
        >
          <div
            class="PodBrowser-drawer__article"
          >
            <button
              class="PodBrowser-button PodBrowser-button--text"
            >
              <span
                class="PodBrowser-icon-cancel"
              />
               Close
            </button>
          </div>
          <section
            class="PodBrowser-headerSection"
          >
            <h3
              class="PodBrowser-content-h3"
              data-testid="resource-title"
              title="https://example.com/container/"
            >
              container
            </h3>
          </section>
          <hr
            class="PodBrowser-root"
          />
          <div
            class="PodBrowser-root PodBrowser-root PodBrowser-expanded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
          >
            <div
              aria-disabled="false"
              aria-expanded="true"
              class="PodBrowser-root PodBrowser-root PodBrowser-expanded"
              data-testid="accordion-resource-actions-trigger"
              role="button"
              tabindex="0"
            >
              <div
                class="PodBrowser-content PodBrowser-expanded"
              >
                Actions
              </div>
              <div
                aria-disabled="false"
                aria-hidden="true"
                class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-expanded PodBrowser-edgeEnd"
              >
                <span
                  class="PodBrowser-label"
                >
                  <svg
                    aria-hidden="true"
                    class="PodBrowser-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
                <span
                  class="PodBrowser-root"
                />
              </div>
            </div>
            <div
              class="PodBrowser-container PodBrowser-entered"
              style="min-height: 0px;"
            >
              <div
                class="PodBrowser-wrapper"
              >
                <div
                  class="PodBrowser-wrapperInner"
                >
                  <div
                    role="region"
                  >
                    <div
                      class="PodBrowser-root PodBrowser-accordionDetails"
                    >
                      <ul
                        class="PodBrowser-action-menu"
                      >
                        <li
                          class="PodBrowser-action-menu__item"
                        />
                        <li
                          class="PodBrowser-action-menu__item"
                        >
                          <button
                            class="PodBrowser-action-menu__trigger PodBrowser-action-menu__trigger--danger"
                            data-testid="delete-resource-button"
                            type="button"
                          >
                            Delete
                          </button>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="PodBrowser-root PodBrowser-root PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
          >
            <div
              aria-disabled="false"
              aria-expanded="false"
              class="PodBrowser-root PodBrowser-root"
              data-testid="accordion-resource-details-trigger"
              role="button"
              tabindex="0"
            >
              <div
                class="PodBrowser-content"
              >
                Details
              </div>
              <div
                aria-disabled="false"
                aria-hidden="true"
                class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-edgeEnd"
              >
                <span
                  class="PodBrowser-label"
                >
                  <svg
                    aria-hidden="true"
                    class="PodBrowser-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
                <span
                  class="PodBrowser-root"
                />
              </div>
            </div>
            <div
              class="PodBrowser-container PodBrowser-hidden"
              style="min-height: 0px;"
            >
              <div
                class="PodBrowser-wrapper"
              >
                <div
                  class="PodBrowser-wrapperInner"
                >
                  <div
                    role="region"
                  >
                    <div
                      class="PodBrowser-root PodBrowser-accordionDetails"
                    >
                      <section
                        class="PodBrowser-centeredSection"
                      >
                        <ul
                          class="PodBrowser-root PodBrowser-padding"
                        >
                          <li
                            class="PodBrowser-root PodBrowser-gutters"
                          >
                            <p
                              class="PodBrowser-root PodBrowser-detailText PodBrowser-body1"
                            >
                              Thing Type:
                            </p>
                            <p
                              class="PodBrowser-root PodBrowser-typeValue PodBrowser-detailText PodBrowser-body1"
                            >
                              text/turtle
                            </p>
                          </li>
                        </ul>
                      </section>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Container view renders Not Supported if resource loaded is not a container 1`] = `
<DocumentFragment>
  <div
    data-testid="not-supported"
  >
    <div
      class="PodBrowser-page-header"
    >
      <h1
        class="PodBrowser-page-header__title"
      >
        Resource Not Supported
      </h1>
      <div
        class="PodBrowser-page-header__content"
      />
      <div
        class="PodBrowser-page-header__toolbar"
      />
    </div>
    <div
      class="PodBrowser-container"
    >
      <div
        class="PodBrowser-content"
      >
        <p>
          PodBrowser can only show resources that are Solid containers.
        </p>
        <p>
          <span>
            Navigate to the resource directly (this will navigate away from PodBrowser): 
          </span>
          <a />
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container view renders Not supported if the fetch for container returns 500 1`] = `
<DocumentFragment>
  <div
    data-testid="not-supported"
  >
    <div
      class="PodBrowser-page-header"
    >
      <h1
        class="PodBrowser-page-header__title"
      >
        Resource Not Supported
      </h1>
      <div
        class="PodBrowser-page-header__content"
      />
      <div
        class="PodBrowser-page-header__toolbar"
      />
    </div>
    <div
      class="PodBrowser-container"
    >
      <div
        class="PodBrowser-content"
      >
        <p>
          PodBrowser can only show resources that are Solid containers.
        </p>
        <p>
          <span>
            Navigate to the resource directly (this will navigate away from PodBrowser): 
          </span>
          <a />
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container view renders PodRootLoadError if it fails to fetch resourceInfo for podRoot 1`] = `
<DocumentFragment>
  <div
    data-testid="pod-root-load-error"
  >
    <div
      class="PodBrowser-page-header"
    >
      <h1
        class="PodBrowser-page-header__title"
      >
        Could not load your Pod
      </h1>
      <div
        class="PodBrowser-page-header__content"
      />
      <div
        class="PodBrowser-page-header__toolbar"
      />
    </div>
    <div
      class="PodBrowser-container"
    >
      <p>
        PodBrowser encountered an error while loading your Pod.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container view renders a table 1`] = `
<DocumentFragment>
  <div
    class="PodBrowser-page-header PodBrowser-files-container-header"
  >
    <h1
      class="PodBrowser-page-header__title"
    >
      Files
    </h1>
    <div
      class="PodBrowser-page-header__content"
    />
    <div
      class="PodBrowser-page-header__toolbar"
    />
  </div>
  <div
    class="PodBrowser-drawer-container"
  >
    <div
      class="PodBrowser-drawer-container__content"
    >
      <div
        class="PodBrowser-sub-header"
      >
        <div
          class="PodBrowser-loading-indicator-container PodBrowser-loading-indicator-container--center"
        >
          <svg
            class="PodBrowser-loading-indicator"
            viewBox="22 22 44 44"
          >
            <circle
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
        <div
          class="PodBrowser-actions-container"
        >
          <a
            class="PodBrowser-page-header__action"
            data-testid="folder-details-button"
            href="/resource/[iri]?action=details&resourceIri="
          >
            Folder Details
          </a>
          <button
            class="PodBrowser-page-header__action"
            data-testid="add-folder-button"
            type="button"
          >
            Create Folder
          </button>
          <label
            class="PodBrowser-page-header__action"
            data-testid="upload-file-button"
            for="upload-file-input"
            tabindex="0"
          >
            Upload File
            <input
              data-testid="upload-file-input"
              id="upload-file-input"
              style="display: none;"
              type="file"
            />
          </label>
        </div>
      </div>
      <table
        class="PodBrowser-table"
        role="table"
      >
        <thead
          class="PodBrowser-table__header"
        >
          <tr
            class="PodBrowser-table__header-row"
            role="row"
          >
            <td
              class="PodBrowser-table__header-cell PodBrowser-table__header-cell--align-center PodBrowser-table__header-cell--width-preview"
              colspan="1"
              role="columnheader"
            >
                
            </td>
            <td
              class="PodBrowser-table__header-cell PodBrowser-table__header-cell--align-center PodBrowser-table__header-cell--width-preview"
              colspan="1"
              role="columnheader"
            >
                
            </td>
            <td
              class="PodBrowser-table__header-cell"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              File 
            </td>
            <td
              class="PodBrowser-table__header-cell"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              Type 
            </td>
          </tr>
        </thead>
        <tbody
          class="PodBrowser-table__body"
          role="rowgroup"
        >
          <tr
            class="PodBrowser-table__body-row PodBrowser-table__body-row--selectable"
            tabindex="0"
          >
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <button
                class="PodBrowser-button PodBrowser-button--text PodBrowser-bookmark-button"
                data-testid="bookmark-button"
                disabled=""
                type="button"
              >
                <i
                  alt="bookmark button"
                  aria-label="bookmark"
                  class="PodBrowser-icon-star PodBrowser-bookmark-icon-unselected"
                />
              </button>
            </td>
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <i
                class="PodBrowser-icon-file PodBrowser-resource-icon"
              />
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              inbox
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              Resource
            </td>
          </tr>
          <tr
            class="PodBrowser-table__body-row PodBrowser-table__body-row--selectable"
            tabindex="0"
          >
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <button
                class="PodBrowser-button PodBrowser-button--text PodBrowser-bookmark-button"
                data-testid="bookmark-button"
                disabled=""
                type="button"
              >
                <i
                  alt="bookmark button"
                  aria-label="bookmark"
                  class="PodBrowser-icon-star PodBrowser-bookmark-icon-unselected"
                />
              </button>
            </td>
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <i
                class="PodBrowser-icon-file PodBrowser-resource-icon"
              />
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              private
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              Resource
            </td>
          </tr>
          <tr
            class="PodBrowser-table__body-row PodBrowser-table__body-row--selectable"
            tabindex="0"
          >
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <button
                class="PodBrowser-button PodBrowser-button--text PodBrowser-bookmark-button"
                data-testid="bookmark-button"
                disabled=""
                type="button"
              >
                <i
                  alt="bookmark button"
                  aria-label="bookmark"
                  class="PodBrowser-icon-star PodBrowser-bookmark-icon-unselected"
                />
              </button>
            </td>
            <td
              class="PodBrowser-table__body-cell PodBrowser-table__body-cell--align-center PodBrowser-table__body-cell--width-preview"
            >
              <i
                class="PodBrowser-icon-file PodBrowser-resource-icon"
              />
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              note.txt
            </td>
            <td
              class="PodBrowser-table__body-cell"
            >
              Resource
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <section
      class="PodBrowser-drawer"
    >
      <div
        class="PodBrowser-drawer__content"
      >
        <div
          class="PodBrowser-drawer__content-wrapper"
        >
          <div
            class="PodBrowser-drawer__article"
          >
            <button
              class="PodBrowser-button PodBrowser-button--text"
            >
              <span
                class="PodBrowser-icon-cancel"
              />
               Close
            </button>
          </div>
          <section
            class="PodBrowser-centeredSection"
          >
            <h3
              class="PodBrowser-content-h3"
              title=""
            >
              null
            </h3>
          </section>
          <hr
            class="PodBrowser-root"
          />
          <div
            class="PodBrowser-loading-indicator-container PodBrowser-loading-indicator-container--center"
          >
            <svg
              class="PodBrowser-loading-indicator"
              viewBox="22 22 44 44"
            >
              <circle
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
